app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,23 @@
+class MessagesController < ApplicationController
+
+  def index
+    @messages = Message.broadcast
+  end
+
+  def create
+    if params[:goal_id]
+      goal = Goal.find_by(id: params[:goal_id])
+    else
+      goal = Pledge.find_by(id: params[:pledge_id]).goal
```

Comment:
An announcement is made when you make a pledge as well, but a pledge comes in with a goal_id in the params as well as the pledge id, so the conditional is not necessary. Removed.